### ISSUE TITLE
Paypal keeps informing me that IPN failed

### ISSUE THREAD

**Issue by donjakobo** [snowplow-archive/codeigniter-paypal-ipn#11]
I've started using your fantastic library, and testing works just fine, I see transactions through the log database and for orders and order items. The API on my side is fully accessible and CSRF is disabled for that URI.

However, Paypal's IPN history keeps notifying me about failing!?  I'm not sure how to debug this now as Paypal says error 500 was thrown, my CI logs don't show any errors, and I see successful entries for logs and IPN's come through just fine.  I'm assuming 'something' is failing somewhere during the response from paypal, maybe I don't respond back? 

Any ideas where to start with this?  I am using a simple IPN response from a web payment form against paypal.

**Comment by donjakobo** [comment]
I ended up debugging further, and I think I discovered the cause, which I must say is just my dumb ignorance and overlooking the issue.  Furthermore from this I would like to propose a solution so by default the error is not thrown until someone enables it.

The cause of my error 500, and failure with Paypal, was the following:

```
// Now construct the email
$emailBody = $this->smarty->view('confirmation_email.tpl', $data, TRUE); // You'll have to create your own email template using Smarty, Twig or similar
```

With the email portion commented out (for now) the IPN responds just fine, and Paypal is showing it working 100%,  I'm not sure how before in sandbox testing this wasn't an issue, maybe something about paypals checking of the IPN response doesn't match production.

Anyway, for now it works fine, I have 'resent' the failed IPNs (eventho I had logs of them as paid), seems fine for now. Will re-open this issue if it persists.
